listenbrainz_spark/recommendations/recommend.py

```diff
@@ -32,203 +25,135 @@ def load_model(path):
-            top_artists_candidate_set (dataframe): Dataframe containing recording ids of top artists.
-            similar_artists_candidate_set (dataframe): Dataframe containing recording ids of similar artists.
-            mapped_listens (dataframe): Dataframe with all the columns/fields that a typical listen has.
+            top_artists_candidate_set: Dataframe containing recording ids that belong to top artists.
+            similar_artists_candidate_set: Dataframe containing recording ids that belong to similar artists.
 
         Returns:
-            recommendations (dict): Dictionary can be depicted as:
-                {
-                    'user_name 1': {
-                        'time': 'xx.xx',
-                        'top_artists_recordings': [
-                            ('xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx'),
-                            ...
-                            ('xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx')
-                        ]
-                        'similar_artists_recordings' : [
-                            ('xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx'),
-                            ...
-                            ('xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx')
-                        ]
-                    }
-                }
+            messages (list): user recommendations.
     """
-    recommendations = defaultdict(dict)
-    for user_name in user_names:
-        try:
-            t0 = time()
-            user_recommendations = recommend_user(user_name, model, recordings_df, users_df, top_artists_candidate_set,
-                similar_artists_candidate_set, mapped_listens)
-            user_recommendations['time'] = '{:.2f}'.format((time() - t0) / 60)
-            current_app.logger.info('Recommendations for "{}" generated'.format(user_name))
-            recommendations[user_name] = user_recommendations
-        except IndexError:
-            current_app.logger.error('{} is new/invalid user.'.format(user_name))
-    return recommendations
-
-def get_recommendation_html(recommendations, time_, best_model_id, ti):
-    """ Prepare and save recommendation HTML.
+    messages = []
+    current_app.logger.info('Generating recommendations...')
+
+    # active users in the last week/month.
+    # users for whom recommendations will be generated.
+    users_df = top_artists_candidate_set.select('user_id', 'user_name').distinct()
+
+    for row in users_df.collect():
+        user_name = row.user_name
+        user_id = row.user_id
+
+        user_recommendations_top_artist, user_recommendations_similar_artist = get_recommendations_for_user(
+                model, user_id,
+                user_name, recordings_df,
+                top_artists_candidate_set,
+                similar_artists_candidate_set
+        )
+
+        messages.append({
+            'musicbrainz_id': user_name,
+            'type': 'recommendations',
```

Comment:
this name can probably be more specific.

Reply:
Want to suggest something? I could not think of anything else.

Reply:
Want to suggest something? I could not think of anything else.

Reply:
`cf_recording_recommendations` at least, i assume eventually we'll have other recommendations coming in from the cluster too.

Reply:
Thanks.